documentation/development-docs/should-process-confirm-impact.md

```diff
@@ -0,0 +1,119 @@
+## ShouldProcess
+
+From [Cmdlet Attribute Declaration](https://msdn.microsoft.com/en-us/library/system.management.automation.cmdletcommonmetadataattribute.supportsshouldprocess(v=vs.85).aspx):
+
+> _All commands that change the system should include the **SupportsShouldProcess** keyword when declaring the attribute. If the ShouldProcess method returns true, the command should perform its action. If the ShouldProcess method returns false, the action should not be taken._
+> 
+> _If a command supports the ShouldProcess method, the **Confirm** and **WhatIf** parameters are added to the command by the Windows PowerShell runtime._
+
+```cs
+[Cmdlet(VerbsCommon.Remove, "Foo", SupportsShouldProcess = true)]
+public class RemoveFoo
+{
+    ...
+
+    public override void ExecuteCmdlet()
+    {
+        if (ShouldProcess(target, actionMessage))
+        {
+            // make changes
+        }
+    }
+}
+```
+
+By default, `ShouldProcess` interacts with the cmdlet `ConfirmImpact` and the user's `$ConfirmPreference` setting to decide whether to prompt the user before continuing cmdlet processing. The `$ConfirmPreference` determines the level at which confirmation automatically occurs, and no prompt is shown. If the `ConfirmImpact` specified in a cmdlet is at or below the level of `$ConfirmPreference`, then processing occurs automatically without displaying a prompt. Since both `ConfirmImpact` and `$ConfirmPreference` are set by default to `Medium`, this means that, by default, no confirmation prompt is shown and the cmdlet executes normally.
+
+PowerShell has several tools that allow users to explore and control what happens during execution of cmdlets, and this change in implementation allows them to use these tools.
+
+Users who specify the `-Confirm` parameter on the command prompt will automatically be prompted for **any** call to `ShouldProcess`, and can decide whether to continue processing. Users can increase or decrease their `$ConfirmPreference` level to decide which kinds of changes will be confirmed automaticlaly without user interaction.
+
+Users can specify the `-WhatIf` parameter to see all the `ShouldProcess` prompts that would occur during exeuction of a cmdlet, without actually making any changes.
+
+Some cmdlets required additional confirmation. For example, if a cmdlet would destroy existing resources in some circumstances, the cmdlet might detect that condition and prompt the user to verify before continuing. Overwriting an existing resource during resource creation, overwriting a file when downloading data, deleting a resource that is currently in use, or deleting a container that contains additional resources are all example of this pattern. To implement additional confirmation, and allow scripts to opt out of additional prompts, the above pattern is enhanced with calles to `ShouldContinue()` and the `Force` parameter:
```

Comment:
Typo:  enhanced with **calles** to `ShouldContinue()` and
Fixed: enhanced with **calls** to `ShouldContinue()` and